qml/SportSelectPopUp.qml

```diff
@@ -18,10 +18,11 @@ Dialog {
       }
    }
    Button {
-      text:i18n.tr("After")
+      text:i18n.tr("Cancel")
```

Comment:
iirc this button was meant to let the user chose the type of sport at the end of the activity. Maybe "Later" is more appropriate that "After", but I don't think it should be "Cancel".
But again, that is only from what I remember, unfortunately I don't have a UT device atm to check this.
If the intention of this button is to record an activity but without choosing the sport type, `newrunEdge.collapse()` shouldn't be added in the action performed at the click of this button.

Reply:
Your "from memory" interpretation does make sense. 

This dialog can be reached in two places:

1. press bottom edge to start a new activity: the dialog is displayed
2. in map view, when an activity was chosen, there is an icon in the header with the activity symbol, pressing that does open the dialog too

For 2. "cancel" would make sense. You do have an activity type set, and you can either select another one or cancel selection mode.

For 1. "it depends". We can have two possible scenarios, both would make sense.

a) The way it is currently designed, "choose later" would be the best wording in my opinion. You press the button and go to map screen where you can record an activity without having selected a type.
This would not "force" a user to select an activity type.

b) My version is a consistent second option. If we name it "Cancel" and use the collapse, then it does cancel starting an activity. So the user is forced to pick a type when starting a new activity.

Codewise going with 1. b) is the easiest option. The button is always labelled "Cancel". Although then 2. will never occur since the icon is not shown. We could show a generic icon there if wanted though.

While recording an activity, there is no indication or additional value I can see for having an activity type selected earlier - except the icon in the header where you can change that activity type.
When stopping an activity, the user always gets asked as which type the activity should be stored. Only the preselection is done.

So the simplest solution is to let a user just start an activity, no prompt for type. When storing the activity, the user is asked for name and type. Then this dialog is not needed at all. This is the least complexity.

If we want to keep it, and we stick with the 1 a) implementation, it should name the button dependent on the usecase. "Cancel" when opened from header and "Choose later" when opened at the start of a new activity.

I hope my words do make some sense to someone not having the app to check.

Reply:
Yeah, thay do make sense, thanks for the detailed explanaiton.
I liked the idea of choosing the sport at the beginning because if you do something tiring you have a fresher mind and later maybe you just want to save the activity without thining too much and just rest. Also, this was my experience with different phone apps and dedicated devices too (watches, etc...)
Is the icon in the header always visible or only if the activity type hasn't been choosen yet? Because if you alredy chose it, then it make sense to have the cancel button, but otherwise if you tap on the icon without having already selected the type and you still don't want to select it, I think it makes sense to have "choose later" there too. And this could be switched easily in the popup looking at the option selector and see if its current index is valid or not.

Reply:
The icon in the header is only visible if you DID select a type. So it would be a confirm/edit/change whatever action. Cancel seems appropriate there. For the other action I would go for "Choose later" for the label.

I removed the collapse action and made a conditional label text. That seems to fit now much better.